File: javascript/easy/palindromeCheck.js
// input: non-empty string
// output: boolean representing if the input is a palindrome

function palindromeCheck(string) {
  for (let i = 0; i < string.length / 2; i++) {
    if (string[i] !== string[string.length - 1 - i]) return false;
  }
  return true;
}

module.exports = palindromeCheck;

File: python/easy/palindrome_check.py
# string / easy
# input: non-empty string
# output: boolean representing if string is a palindrome

# O(n) time
def is_palindrome(string):
    for i in range(len(string)):
        if i >= len(string) - 1 - i:
            break
        if string[i] != string[len(string) - 1 - i]:
            return False
    return True
